lib/elixir/lib/module/types/descr.ex

```diff
@@ -202,4 +232,91 @@ defmodule Module.Types.Descr do
         (mask &&& val) !== 0,
         do: {type, [], []}
   end
+
+  defp bitmap_count_top_types(val), do: Integer.digits(val, 2) |> Enum.sum()
+
+  ## Atoms
+
+  # The atom component of a type consists of pairs `{MapSet.t(atom()), boolean()}`.
+  # If the boolean is true, the type represents the union of the atoms in the set.
```

Comment:
The wording below is `{s1, neg1}`, it seems misleading that `neg1 = true`, means non-negated union of `s1` and `neg1 = false` means negated union of `s1`. Shouldn't this be the other way round, or am I misinterpreting? :D

Reply:
You are getting it right! I should either switch it, find a better name for the variable, or implement @sabiwara's suggestion. I'm trying out the latter right now.